CMakeLists.txt

```diff
@@ -92,6 +92,7 @@ if(CMAKE_COMPILER_IS_GNUCC OR "${CMAKE_CXX_COMPILER_ID}" MATCHES "(Apple)?Clang"
     add_cxx_flag_if_supported(-Wno-narrowing)
     add_cxx_flag_if_supported(-Wno-format)
     add_cxx_flag_if_supported(-Werror)
+    add_cxx_flag_if_supported(-Wunused-variable)
```

Comment:
It would be better to move `-Wunused-variable` up, to go with `-Wmisleading-indentation`, given that `-Werror` is not the same _type_ of warning option as `-Wunused-variable` or as `-Wno-format`.

The sequence of options previously was:
* `-W*` options to enable warnings / categories thereof
* `-Wno-*` options to disable warnings
* `-Werror` to turn warnings into errors
* `-Wno-error=*` to disable turning warnings into errors for specific warnings / categories thereof

I think it would be good to preserve this ordering.

Reply:
Makes sense, done.